microprofile/lra/jax-rs/src/main/java/io/helidon/microprofile/lra/CoordinatorLocatorService.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.microprofile.lra;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.DeploymentException;
+import javax.inject.Inject;
+
+import io.helidon.common.serviceloader.HelidonServiceLoader;
+import io.helidon.lra.coordinator.client.CoordinatorClient;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import static io.helidon.lra.coordinator.client.CoordinatorClient.CONF_KEY_COORDINATOR_TIMEOUT;
+import static io.helidon.lra.coordinator.client.CoordinatorClient.CONF_KEY_COORDINATOR_TIMEOUT_UNIT;
+import static io.helidon.lra.coordinator.client.CoordinatorClient.CONF_KEY_COORDINATOR_URL;
+
+
+@ApplicationScoped
+class CoordinatorLocatorService {
+
+    @Inject
```

Comment:
Please use constructor injection if possible.

Reply:
Also the module name seems to say this is JAX-RS specific mdoule, yet this class integrates with CDI

Reply:
It is CDI extension registering set of jax-rs filters, we have agreed on the name in previous call in case we decide to extend LRA to support non-jaxrs beans in the future.
https://github.com/oracle/helidon/pull/3016#discussion_r683339283